subsys/bluetooth/services/fast_pair/fp_crypto/fp_crypto_psa.c

```diff
@@ -0,0 +1,253 @@
+	psa_set_key_bits(&key_attr, len * CHAR_BIT);
+
+	psa_status_t status = psa_import_key(&key_attr, data, len, &key_id);
+
+	psa_reset_key_attributes(&key_attr);
+
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_import_key failed (err: %d)", status);
+		key_id = PSA_KEY_ID_NULL;
+	}
+
+	return key_id;
+}
+
+int fp_crypto_hmac_sha256(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *hmac_key,
+			  size_t hmac_key_len)
+{
+	int err = 0;
+	size_t olen = 0;
+	psa_key_id_t hmac_key_id;
+	psa_status_t status;
+
+	hmac_key_id = import_hmac_sha256_key(hmac_key, hmac_key_len);
+	if (hmac_key_id == PSA_KEY_ID_NULL) {
+		LOG_ERR("import_hmac_sha256_key failed");
+		return -EIO;
+	}
+
+	status = psa_mac_compute(hmac_key_id, PSA_ALG_HMAC(PSA_ALG_SHA_256), in, data_len,
+				 out, FP_CRYPTO_SHA256_HASH_LEN, &olen);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_mac_compute failed (err: %d)", status);
+		err = -EIO;
+	}
+
+	if (!err && (olen != FP_CRYPTO_SHA256_HASH_LEN)) {
+		LOG_ERR("Invalid psa_mac_compute output length: %zu", olen);
+		err = -EIO;
+	}
+
+	status = psa_destroy_key(hmac_key_id);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_destroy_key failed (err: %d)", status);
+	}
+
+	return err;
+}
+
+static psa_key_id_t import_aes128_key(const uint8_t *data)
+{
+	static const size_t len = FP_CRYPTO_AES128_KEY_LEN;
+
+	psa_key_id_t key_id = PSA_KEY_ID_NULL;
+	psa_key_attributes_t key_attr = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attr, PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT);
+	psa_set_key_lifetime(&key_attr, PSA_KEY_LIFETIME_VOLATILE);
+	psa_set_key_algorithm(&key_attr, PSA_ALG_ECB_NO_PADDING);
+	psa_set_key_type(&key_attr, PSA_KEY_TYPE_AES);
+	psa_set_key_bits(&key_attr, len * CHAR_BIT);
+
+	psa_status_t status = psa_import_key(&key_attr, data, len, &key_id);
+
+	psa_reset_key_attributes(&key_attr);
+
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_import_key failed (err: %d)", status);
+		key_id = PSA_KEY_ID_NULL;
+	}
+
+	return key_id;
+}
+
+static int aes128_ecb_crypt(uint8_t *out, const uint8_t *in, const uint8_t *k, bool encrypt)
+{
+	int err = 0;
+	size_t olen = 0;
+	psa_key_id_t key_id;
+	psa_status_t status;
+
+	key_id = import_aes128_key(k);
+	if (key_id == PSA_KEY_ID_NULL) {
+		LOG_ERR("import_aes128_key failed");
+		return -EIO;
+	}
+
+	if (encrypt) {
+		status = psa_cipher_encrypt(key_id, PSA_ALG_ECB_NO_PADDING,
+					    in, FP_CRYPTO_AES128_BLOCK_LEN,
+					    out, FP_CRYPTO_AES128_BLOCK_LEN, &olen);
+	} else {
+		status = psa_cipher_decrypt(key_id, PSA_ALG_ECB_NO_PADDING,
+					    in, FP_CRYPTO_AES128_BLOCK_LEN,
+					    out, FP_CRYPTO_AES128_BLOCK_LEN, &olen);
+	}
+
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_cipher_%scrypt failed (err: %d)", encrypt ? "en" : "de", status);
+		err = -EIO;
+	}
+
+	if (!err && (olen != FP_CRYPTO_AES128_BLOCK_LEN)) {
+		LOG_ERR("Invalid psa_cipher_%scrypt output length: %zu",
+			encrypt ? "en" : "de", olen);
+		err = -EIO;
+	}
```

Comment:
You could create a helper function aes128_ecb_crypt_process to simplify error-handling logic. Also in other functions

```
crypto_operation_funct(...)
{
	import_key(...);
	crypto_operation_process(...);
        destroy_key(...);
}

Reply:
See https://github.com/nrfconnect/sdk-nrf/pull/14733/files/9621f79955d0c2488f76c8ee60713661601d9c61#r1555826805

Since key import/destroy functions are probably going to be removed, I would desist from modifying this part for now. Current approach was taken to simplify transition to handling keys on secure side.

Reply:
all the more reason to have separated like this. The current error handling won't be needed in this case as we won't have to call destroy key function at the end

Reply:
I will add a small wrapper for PSA crypto operation processing itself. It may be useful in the future if we want to introduce PSA compatible Fast Pair crypto APIs.